### DISCUSSION TITLE
Week 1. Orientation and Fundamentals

### DISCUSSION THREAD

**Discussion by choonho** [cloudforet-io/community#133] [Mentorship]
Hello All,

There are many person in this Mentorship from different country.
So we are going to discuss in online.

# Week 1. consist of three agenda.

**Date: 2023-11-27 ~ 2023-12-10 (Extended)**

# Link for Google Calendar (LXF Mentorship)
Anyone who has problem to solve the issue. Make 1:1 reservation on Google Calendar. There are lots of slot.
[Google Calendar](https://calendar.google.com/calendar/u/0?cid=MTdlYzU2MjFjNDlhN2Y3ZGFlMzc1NWJhM2EwNjhkZmNhNDA2NTY0OGMxYTM4ZGZmMWE3MWViZjlmMGIwZDU1NUBncm91cC5jYWxlbmRhci5nb29nbGUuY29t)

How to use: https://github.com/orgs/cloudforet-io/discussions/136

## Survey
* [Survey](https://docs.google.com/forms/d/e/1FAIpQLSfkmbIPuPc5AyN2lDcBCmhxjD2lt_W-YS_Llatxo40MfXSdxA/viewform?usp=sf_link)

Please answer the survey, I will check your status.

## Introduction
* [Introduction to Cloudforet, Googleslide](https://docs.google.com/presentation/d/1iJvxHBB79xADKaHt2HfjDVSwd1m_COxiURx_M2Vo1uA/edit?usp=sharing)
* [Introduction to Cloudforet, Youtube](https://youtu.be/FBNfQg1vCuc)

## Installation
* [Installation of Cloudforet with minikube](https://cloudforet.io/docs/setup_operation/quick_install/)

### submit a file in LXF

> png, jpg is not allowed to upload. Please convert png/jpg to PDF format.

After installation, please upload your login screen like
![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/aa3c5ab1-815b-459f-bf2c-64364684fe8b)

In the login window, the installed version should be "Version 1.12.X" not "Version 2.0.X".

Based on this instruction, please install Cloudforet with Minikube.
If you have any question about installation, ask at https://github.com/orgs/cloudforet-io/discussions/134

**Comment by choonho** [comment]
In your mentee Task page, there will be three same task "Week 1".

All three is same. There is no way to delete two same task. Sorry :)

![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/84f9e39a-aaa1-4a02-8462-4afd0d16464d)

**Reply by py-dev-nandini-12** [reply]
Hi choonho,
Beside the tasks there is a place for uploading.
Can you please clarify?

**Reply by choonho** [reply]
I will update Task like
![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/45de12df-5129-40f0-b880-eeb09b90d351)

The pending task has "mentee submit a file"

**Comment by choonho** [comment]
The correct login screen looks like

Version: 1.12.12

![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/d602d1b6-8a4f-41fa-855c-a9b6c0ed5db6)

**Comment by Gaurav122000** [comment]
Hi,
I'm setting up the project but having some issue on step 6
(Create the values.yaml file and edit the values. There is only one item that need to be updated.) how to do this

**Reply by ATREAY** [reply]
I'm facing the same problem, @choonho. It would be really helpful if you could give us some guidance.

Below I've attached a screenshot for reference:

![image](https://github.com/cloudforet-io/cloudforet/assets/66585295/72528ba1-722f-401a-b126-f3befa56468a)

**Reply by choonho** [reply]
If you run command ***kubectl get pod -n spaceone***. The initialize-spaceone POD may not ***complete***

like 
~~~
identity-6d7bbb678f-b5ptf                  1/1     Running     0          24m
initialize-spaceone-fsqen-74x7v            0/1     Completed   0          98m
inventory-64d6558bf9-v5ltj                 1/1     Running     0  
~~~

I think you logs the POD while initializing (ex. downloading Docker image)

**Reply by Gaurav122000** [reply]
@choonho I completed this step already but what to do in 6th step?

**Reply by Ayush-Sharma410** [reply]
@Gaurav122000 Hey, after implementing this command i repeated the steps in step 6. It didn't work the first time but after some time i did  it again and it worked fine.

**Reply by Gaurav122000** [reply]
@Ayush-Sharma410 Command is working but my doubt is what values need to be updated in this step (Create the values.yaml file and edit the values. There is only one item that needs to be updated).

**Reply by py-dev-nandini-12** [reply]
@Ayush-Sharma410 the values.yaml file will be exactly as given only you need to provide your token which you have received.

**Reply by Ayush-Sharma410** [reply]
@Gaurav122000 You have to create a file in the directory where you are executing all the commands. For the content inside refer this screenshot.
![Screenshot 2023-11-28 205604](https://github.com/cloudforet-io/cloudforet/assets/98555526/d122576f-69e2-4bcf-843d-c068c57bc076)
Just replace the token with the token that you got in your terminal.

**Reply by Gaurav122000** [reply]
@Ayush-Sharma410 After completed step 6 I ran "kubectl delete po -n spaceone -l app.kubernetes.io/instance=cloudforet" then it start to show ( pod "statistics-5fd4bcb4cc-jww95" deleted
pod "statistics-scheduler-687fd4ccb-hrxhr" deleted
pod "statistics-worker-7d97f6cf4d-c9fgt" deleted
pod "supervisor-scheduler-84b8b4b659-kwkzw" deleted
W1129 13:57:29.367509   13952 reflector.go:458] vendor/k8s.io/client-go/tools/watch/informerwatcher.go:146: watch of *unstructured.Unstructured ended with: an error on the server ("unable to decode an event from the watch stream: http2: client connection lost") has prevented the request from succeeding
W1129 13:57:40.911612   13952 reflector.go:535] vendor/k8s.io/client-go/tools/watch/informerwatcher.go:146: failed to list *unstructured.Unstructured: Get "https://127.0.0.1:51346/api/v1/namespaces/spaceone/pods?fieldSelector=metadata.name%3Dboard-6785c9fb6d-8hqng&resourceVersion=4870": net/http: TLS handshake timeout
I1129 13:57:40.911612   13952 trace.go:236] Trace[273199767]: "Reflector ListAndWatch" name:vendor/k8s.io/client-go/tools/watch/informerwatcher.go:146 (29-Nov-2023 13:57:30.909) (total time: 10001ms): 
and stuck at here not moving forward for 5 to 6 min. so I pressed ctrl+c,
Now when I'm running "kubectl get pod -n spaceone" this it showing Unable to connect to the server: net/http: TLS handshake timeout

**Reply by ATREAY** [reply]
Now I'm also getting the same error after running the command @Gaurav122000 mentioned.

**Reply by choonho** [reply]
@Gaurav122000 You finished until step 6. Port-forwarding is the problem of Windows.
I will check you status **completed**

**Comment by madguy02** [comment]
@choonho i have installed and setup cloudforet locally, i have uploaded the screenshot(in pdf format, as the submit button was not taking in png/jpg), please verify or let me know if something needs to be updated.

Also for reference i am adding the screenshot here as well
![Screenshot from 2023-11-27 23-50-13](https://github.com/cloudforet-io/cloudforet/assets/8217234/849288b8-3fe1-4da9-b9b3-948127c2618c)

**Reply by choonho** [reply]
Thank you Manish. Please upload screenshot by PDF format.

**Comment by Ayush-Sharma410** [comment]
@choonho I am on the last step but it is giving 404 error. It would be of great help if you look into this.
![Screenshot 2023-11-28 010854](https://github.com/cloudforet-io/cloudforet/assets/98555526/02859665-9380-4820-a622-9786a8aae296)

**Reply by py-dev-nandini-12** [reply]
@Ayush-Sharma410 I guess you need to use all three commands mentioned at the last step in cli.

**Reply by Ayush-Sharma410** [reply]
@py-dev-nandini-12 I tried to run the first command without '&' sign otherwise it was giving an error.  Also i guess i can run one command at a time and have to stop the port before starting another one. Please correct me if i am wrong. Thank you

**Reply by py-dev-nandini-12** [reply]
@Ayush-Sharma410 yes you are right, but I used the & sign in the command as well.

**Reply by Ayush-Sharma410** [reply]
@py-dev-nandini-12 did you use windows powershell?

**Reply by py-dev-nandini-12** [reply]
Nope I used terminal

**Reply by Ayush-Sharma410** [reply]
i am getting this in terminal
![image](https://github.com/cloudforet-io/cloudforet/assets/98555526/359330a2-6410-4ecd-b51e-ec1d0886ee17)

**Reply by py-dev-nandini-12** [reply]
Did you go to the server mentioned bellow, u need to copy and paste that in the browser http://127.0.0.1:8080/

**Reply by Ayush-Sharma410** [reply]
yes done, still not working

**Reply by py-dev-nandini-12** [reply]
Is it still showing 404 ? One of the commands should keep running in your terminal and then check the url, mine worked.

**Reply by Ayush-Sharma410** [reply]
It is showing this error : 
error: '0.0.0.0' is not a valid IP

**Reply by Ayush-Sharma410** [reply]
It worked 
For windows powershell use the following commands:
 Start-Process kubectl -ArgumentList "port-forward -n spaceone svc/console 8080:80 --address=0.0.0.0" -NoNewWindow -PassThru
 Start-Process kubectl -ArgumentList "port-forward -n spaceone svc/console-api 8081:80 --address=0.0.0.0" -NoNewWindow -PassThru
 Start-Process kubectl -ArgumentList "port-forward -n spaceone svc/console-api-v2-rest 8082:80 --address=0.0.0.0" -NoNewWindow -PassThru

**Reply by choonho** [reply]
This is first try in Windows OS.
Great job! If you post your history, I will link your post in Docs.

**Reply by Ayush-Sharma410** [reply]
Sure sir 
Just wanted to ask if i should post the whole history starting from step 1 or only the parts where i encountered an error in windows powershell and debugged it?

**Reply by choonho** [reply]
If you have your own blog, how about making a new post like "installation of Cloudforet in Windows". Since most of developer does not know how to install docker, minikube, powershell in Windows.

Then I will link your post like

# Reference
[Full Installation Guide](https://github.com/cloudforet-io/charts)
[Discuss channel](https://github.com/orgs/cloudforet-io/discussions/124)
[Cloudforet in Windows, Blog](https://cloudforet.io)

**Reply by ATREAY** [reply]
Hey @choonho, @Ayush-Sharma410, or anyone who has managed to set up Cloudforet locally, I'm having the same issue. I gave Ayush's suggestion a shot, but now I'm facing this new error:

Here is the screenshot of the error:-
![image](https://github.com/cloudforet-io/cloudforet/assets/66585295/4d25a911-5a2b-4ff7-ad93-ca6859cae335)

**Reply by Ayush-Sharma410** [reply]
@choonho Sure sir i'll make a blog and share with you asap.

**Reply by Ayush-Sharma410** [reply]
@ATREAY Can you share with the screenshot of the commands you executed before this.

**Reply by ATREAY** [reply]
Before this, I ran these commands:

```powershell
Start-Process kubectl -ArgumentList "port-forward -n spaceone svc/console 8080:80 --address=0.0.0.0" -NoNewWindow -PassThru
Start-Process kubectl -ArgumentList "port-forward -n spaceone svc/console-api 8081:80 --address=0.0.0.0" -NoNewWindow -PassThru
Start-Process kubectl -ArgumentList "port-forward -n spaceone svc/console-api-v2-rest 8082:80 --address=0.0.0.0" -NoNewWindow -PassThru
```

And the commands mentioned in step 6.

**Reply by ATREAY** [reply]
Any updates @choonho or @Ayush-Sharma410?

**Reply by Tyagiquamar** [reply]
@ATREAY 
how did you copleted this step 
>kubectl get pod -n spaceone
Unable to connect to the server: net/http: TLS handshake timeout

**Reply by ATREAY** [reply]
I think you need to restart docker and then restart minikube using the 1st command.

**Reply by Tyagiquamar** [reply]
@ATREAY tried that, but very few container running 
>kubectl get pod -n spaceone by this command, may be 3 or 4

**Reply by ATREAY** [reply]
Hey @choonho, @Ayush-Sharma410, or anyone who has managed to set up cloudforet locally, could you please help me out with the issue I'm facing?

**Reply by ATREAY** [reply]
Hey @Tyagiquamar, in my situation, not all containers started right away. I waited for about 5 minutes, and then everything started running smoothly.

**Reply by choonho** [reply]
@Tyagiquamar , after "kubectl get pod -n spaceone", list the pod which is not ***Running***. Try "docker pull" manually. May be you have very slow internet connection.

For example,
~~~
kubectl get pod -n spaceone

NAME                                      READY   STATUS             RESTARTS      AGE
board-5746fd9657-vtd45                    1/1     Initializing            0             57s
config-5d4c4b7f58-z8k9q                   1/1     Running            0             58s
console-6b64cf66cb-q8v54                  1/1     Running    
~~~

You can try like
~~~
docker pull cloudforet/board
~~~

Check how much time take?

**Reply by Tyagiquamar** [reply]
@choonho  it shows error this time 
docker pull cloudforet/board
Using default tag: latest
Error response from daemon: Get "https://registry-1.docker.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
maybe because of our college wifi, they most of the times mess-up/block the repository(that is why I am unable to use linux).

wifi speed is 20MB/s
but 15 minutes and only 3 are running, will try to leave it overnight.

is it because my minikube size is 1900mb.

**Reply by Cvr421** [reply]
Hello @choonho @Ayush-Sharma410  i am also getting the  404 error in  browser. The command you have mention for port forwarding . i  have run that in window terminal but still getting same 404 error . Can you please describe more about how you have completed these step . 
Anyone have some thought please feel free to write down. 
![Screenshot 2023-12-01 021659](https://github.com/cloudforet-io/cloudforet/assets/82499697/8701048b-5322-4cb0-bae3-9ab8c9be043a)
Output of window terminal after port forwarding command 

Start-Process kubectl -ArgumentList "port-forward -n spaceone svc/console 8080:80 --address=0.0.0.0" -NoNewWindow -PassThru
Start-Process kubectl -ArgumentList "port-forward -n spaceone svc/console-api 8081:80 --address=0.0.0.0" -NoNewWindow -PassThru
Start-Process kubectl -ArgumentList "port-forward -n spaceone svc/console-api-v2-rest 8082:80 --address=0.0.0.0" -NoNewWindow -PassThru




![Screenshot 2023-12-01 022244](https://github.com/cloudforet-io/cloudforet/assets/82499697/7e59ef5f-5470-4033-b1de-6d033d1e5874)

**Reply by Tyagiquamar** [reply]
@Cvr421 i am also at this stage, did your issue get resolved.

**Reply by Cvr421** [reply]
No @Tyagiquamar . i request @py-dev-nandini-12 @Ayush-Sharma410 to put some thoughts . On how they completed .

**Reply by py-dev-nandini-12** [reply]
@Cvr421 For the last step I ran three commands in terminal one after another and it worked for me, I used  exactly the same commands as mentioned in the documentation.And as per the screenshot above it’s showing handling means the port is working , if you paste the url provided in the browser at this stage you should get to see the login page, once done I have terminated the command.

**Reply by py-dev-nandini-12** [reply]
<img width="1680" alt="Screenshot 2023-12-02 at 11 40 01" src="https://github.com/cloudforet-io/cloudforet/assets/127988967/8aec5ea5-708b-4796-a29d-1edc7837af19">
This was my output in terminal:
<img width="1680" alt="Screenshot 2023-12-02 at 11 40 18" src="https://github.com/cloudforet-io/cloudforet/assets/127988967/5ee16684-3635-4d7d-8bc4-a097fb9b6920">
And the login page, after pasting the given url in the browser.

**Reply by ATREAY** [reply]
Alright, I believe the problem might be with the Windows OS.

**Reply by Cvr421** [reply]
> <img width="1680" alt="Screenshot 2023-12-02 at 11 40 01" src="https://github.com/cloudforet-io/cloudforet/assets/127988967/8aec5ea5-708b-4796-a29d-1edc7837af19">
> This was my output in terminal:
> <img width="1680" alt="Screenshot 2023-12-02 at 11 40 18" src="https://github.com/cloudforet-io/cloudforet/assets/127988967/5ee16684-3635-4d7d-8bc4-a097fb9b6920">
> And the login page, after pasting the given url in the browser.

@py-dev-nandini-12  thanks for replying.Are you on mac right .i think the issue is with window os . I have run the same as mentioned.but nothing worked.

**Reply by Gaurav122000** [reply]
hey @ATREAY did your worked on windows because I also getting 404

**Reply by py-dev-nandini-12** [reply]
@Cvr421 yes on Mac

**Reply by choonho** [reply]
@Cvr421 I created 1:1 google calendar. Please reserve your slot. I will help you.
https://calendar.google.com/calendar/u/0?cid=MTdlYzU2MjFjNDlhN2Y3ZGFlMzc1NWJhM2EwNjhkZmNhNDA2NTY0OGMxYTM4ZGZmMWE3MWViZjlmMGIwZDU1NUBncm91cC5jYWxlbmRhci5nb29nbGUuY29t

**Reply by Cvr421** [reply]
Thanks @choonho

**Reply by Ayush-Sharma410** [reply]
@choonho please check my blog for installation of cloudforet in windows.
Here is the link --> (https://medium.com/@ayushsharma2267410/installation-of-cloudforet-in-windows-8c4a10c9a65f)
Please review and tell if any changes needed.
Thank you

**Comment by choonho** [comment]
# Tips

During mentoring, I will give your additional issues which can be commit with easy.

Issue Tag like "good first issue".

![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/50f0ac31-d0df-4905-90eb-c42e036ec0f7)

Anyone who can fix, please make PR.

| Repository | Issue Number | Status |
| ---              | ---                    |    ---   | 
| docs          | https://github.com/cloudforet-io/docs/issues/169                 |  |
| docs          | https://github.com/cloudforet-io/docs/issues/171                |  |

**Reply by choonho** [reply]
I added two more "good first issue" ticket at https://github.com/cloudforet-io/docs/issues

**Comment by Tyagiquamar** [comment]
I am stuck at step 4 in windows 
>kubectl get pod -n spaceone
Unable to connect to the server: net/http: TLS handshake timeout
![Screenshot (24)](https://github.com/cloudforet-io/cloudforet/assets/104281681/d5de95c1-ec66-4e92-aad0-b87d960058dc)

**Reply by choonho** [reply]
Did you create spaceone namespace. The command is "kubectl get ns". Or there is problem on your minikube.

**Reply by Tyagiquamar** [reply]
yeah, I created a namespace.
How to reset and start everything from minikube.

**Reply by Tyagiquamar** [reply]
@choonho My minukube ram size is 1900mb.
I waited for 10 minutes no one  is starting,
all of the instances saying containercreating.

**Reply by choonho** [reply]
@Tyagiquamar I think the RAM size of minikube is small. That's why I guided ***minikube start --driver=docker --memory=4096mb***.
Can you increate the size?

**Reply by Tyagiquamar** [reply]
Yeah, I resetted the docker and its config file and increases the config file.
what do you think about campus wifi, it also let my PC with DNS problem, shall I use my personal data.

**Reply by choonho** [reply]
@Tyagiquamar If you use campus wifi, I think it will be OK.
Docker has the feature of image pull cache.
If you has problem of slow network.

It is possible to pull docker image manually, before helm install.

~~~
docker pull cloudforet/console:1.12.12
docker pull cloudforet/console-api:1.12.1
docker pull cloudforet/identity:1.12.4
docker pull cloudforet/secret:1.12.0
docker pull cloudforet/repository:1.12.3
docker pull cloudforet/plugin:1.12.0
docker pull cloudforet/config:1.12.0
docker pull cloudforet/inventory1.12.2
docker pull cloudforet/monitoring:1.12.0
docker pull cloudforet/statistics:1.12.0
docker pull cloudforet/cost-analysis:1.12.4
docker pull cloudforet/notification:1.12.0
docker pull cloudforet/board:1.12.0
docker pull cloudforet/file-manager:1.12.1
docker pull cloudforet/dashboard:1.12.0
docker pull cloudforet/console-api-v2:1.12.6
docker pull cloudforet/supervisor:1.12.0
~~~

**Reply by Cvr421** [reply]
@Tyagiquamar Can you try to configure window firewall once . just go to firewall inbound rule and check Docker desktop rule . and make it to allow to all Connection . That might help if you are getting the timeout error.  sometime firewall block the connection between the docker desktop

**Reply by Tyagiquamar** [reply]
@Cvr421 
![Screenshot (26)](https://github.com/cloudforet-io/cloudforet/assets/104281681/c697635f-5908-4f7d-8907-2f23f674d30a)
any tips

**Reply by Cvr421** [reply]
wait until the initialize-spaceone pod is in complete status which are in now init:0/1 which is in initialization state . then go with futher step

**Reply by Tyagiquamar** [reply]
@choonho  @Cvr421 ok, everything is working fine until step 5.
but after running kubectl delete po -n spaceone -l app.kubernetes.io/instance=cloudforet
it deletes a lot of files, then kubectl get pod -n spaceone showed error in some.
can I ignore the delete part

**Reply by Cvr421** [reply]
> @choonho  @Cvr421 ok, everything is working fine until step 5.
> but after running kubectl delete po -n spaceone -l app.kubernetes.io/instance=cloudforet
> it deletes a lot of files, then kubectl get pod -n spaceone showed error in some.
> can I ignore the delete part

First run this command
Kubctl delete pod  -ALL  -n spaceone
And then 
kubectl delete po -n spaceone -l app.kubernetes.io/instance=cloudforet

It mentioned in doc

**Reply by Tyagiquamar** [reply]
@Cvr421 Kubctl delete pod -ALL -n spaceone
And then 
this command did not worked,
 so I skipped it and executed the kubectl delete po -n spaceone -l app.kubernetes.io/instance=cloudforet
now all pods are working, 
but 
![Screenshot (39)](https://github.com/cloudforet-io/cloudforet/assets/104281681/6c48c2d9-92de-4583-b50b-e99116bccd7a)

**Reply by Cvr421** [reply]
@Tyagiquamar I am also stuck at last step . and getting 404 error on web page . i try different things but didn't work.  i request @Ayush-Sharma410  to put some thoughts .

**Reply by Tyagiquamar** [reply]
@Cvr421 Have you successfully installed, I was busy with the exams. If you have installed can you help me

**Reply by choonho** [reply]
@Tyagiquamar I think your problem is related with Windows. Your installation is almost completed without port-forwarding. Let's move on Task 2. I highly recommend you prepare Linux Machine instead of Windows WSL.

**Comment by Tyagiquamar** [comment]
@choonho @Cvr421 @Ayush-Sharma410 I am getting this error 
![Screenshot (33)](https://github.com/cloudforet-io/cloudforet/assets/104281681/a0002efd-6880-42e0-9984-a0d36b6fd112)


because few pods are showing crashLoopbackof 
![Screenshot (32)](https://github.com/cloudforet-io/cloudforet/assets/104281681/0ae78902-b078-4ca2-bf1e-934be947863e)

**Reply by Cvr421** [reply]
@choonho @Ayush-Sharma410 @Tyagiquamar I have also getting same error 404 page . But my all pods are in running status . i think in my case i have issue with port forwarding . And i try lot of things Blog , article but nothing worked. i have also run the command mention by @Ayush-Sharma410 about port forwarding but still getting same error.

**Reply by Cvr421** [reply]
@Tyagiquamar  read the instruction carefully in this article to get the all pods are running state
![image](https://github.com/cloudforet-io/cloudforet/assets/82499697/b1a17547-fe13-4daa-a1c7-3ad9ac5048ec)

**Reply by choonho** [reply]
@Tyagiquamar I created 1:1 google calendar. Please reserve your slot.
https://calendar.google.com/calendar/u/0?cid=MTdlYzU2MjFjNDlhN2Y3ZGFlMzc1NWJhM2EwNjhkZmNhNDA2NTY0OGMxYTM4ZGZmMWE3MWViZjlmMGIwZDU1NUBncm91cC5jYWxlbmRhci5nb29nbGUuY29t

**Reply by SpringWiz11** [reply]
Hello there,

For the application to run correctly we have to make sure that all the pods are running correctly except the scheduler pods. Then the CLI commands for the port forwarding has to be used for running the application.

Here's the trick:- We have 3 CLI commands for the port forwarding :
1. ```kubectl port-forward -n spaceone svc/console 8080:80 --address='0.0.0.0' &```
2. ```kubectl port-forward -n spaceone svc/console-api 8081:80 --address='0.0.0.0' &```
3. ```kubectl port-forward -n spaceone svc/console-api-v2-rest 8082:80 --address='0.0.0.0' &```

Here, I got my application running by running all the 3 CLI commands different terminals. Maybe, you too can try this after making sure the necessary pods are the in the running state.

( I use Macbook pro M2 )

**Reply by Cvr421** [reply]
Hii @SpringWiz11 it is running perfectly fine with linux and mac environment it has the problem with widows environment . But thanks for sharing this information.

**Reply by Ayush-Sharma410** [reply]
@Cvr421 @Tyagiquamar sorry for the late reply. Was your problem solved?

**Reply by Tyagiquamar** [reply]
@Cvr421 No, chohoo asked me to skip, Now i tried to install cloudforet by dual booting manajro with windows, but minikube is not working . Can you help me .

**Reply by Cvr421** [reply]
@Tyagiquamar their is problem with installation on window . try on linux

**Comment by choonho** [comment]
Ih any one has problem to install Cloudforet, I will help 1:1 online. Please check google calendar. I created some time-slot what I can help you. https://calendar.google.com/calendar/u/0?cid=MTdlYzU2MjFjNDlhN2Y3ZGFlMzc1NWJhM2EwNjhkZmNhNDA2NTY0OGMxYTM4ZGZmMWE3MWViZjlmMGIwZDU1NUBncm91cC5jYWxlbmRhci5nb29nbGUuY29t

See https://github.com/orgs/cloudforet-io/discussions/136

**Reply by SammyBloom** [reply]
Thanks. Because I have been on this for some days now and I keep on getting the ErrImagePull for the console-api pod

**Reply by Tyagiquamar** [reply]
@choonho how to get the slots, I tried to join the meeting waiting for host to join.

**Reply by choonho** [reply]
For windows user, let's skip port-forwarding command. If you succeed **step 5. initialization**, installation  of Cloudforet is working.

**Reply by Tyagiquamar** [reply]
@choonho thanks

**Comment by choonho** [comment]
> Thanks. Because I have been on this for some days now and I keep on getting the ErrImagePull for the console-api pod

I noticed some person has ErrorImagePull, this may be your EBS size of EC2 instance is too small. Maybe there is an error message  like "no disk space". Please create EC2 instance with 30GB Disk (default is 8GB)

**Comment by AnsahMohammad** [comment]
@choonho I'm sorrry, i'm a bit late, but i'm still struggling to download the cloudforet.

I'm in step 0 (unable to start minikube)
I am using Ubuntu LTS 22.04
I;m getting the following error :
![image](https://github.com/cloudforet-io/cloudforet/assets/85027826/5edec3c5-24db-4f97-b019-6a423c7a3ad0)

( i missed the message for 1:1 mentoring)

**Reply by choonho** [reply]
@AnsahMohammad Hi, let's make a 1:1 online meet.
How about
1) Tue, 7:30pm(IST)
2) Wed, 7:30pm(IST)

**Reply by AnsahMohammad** [reply]
We can do it on Tuesday (12 December) 7:30pm (IST).  Thank you

**Reply by choonho** [reply]
@AnsahMohammad See 7:30pm(IST) https://meet.google.com/gzb-wkkj-bxs

**Reply by AnsahMohammad** [reply]
@choonho btw, the above issue is solved, but now i'm stuck at step 06. 

`initialize-spaceone-hklma-ccklx            0/1     Init:0/1           0                11m`
```
ansah@lynx:~$ kubectl logs initialize-spaceone-hklma-ccklx -n spaceone
Defaulted container "job-2-initialize-spaceone" out of: job-2-initialize-spaceone, job-1-spacectl-conf-writable (init)
Error from server (BadRequest): container "job-2-initialize-spaceone" in pod "initialize-spaceone-hklma-ccklx" is waiting to start: PodInitializing
```
Unable to get the logs from initialize-spaceone. Does it take time or did i do something wrong. Did till step 05 without any errors

**Reply by choonho** [reply]
Wait until POD is **completed**. initializing means downloading docker image. This takes some times.

**Reply by AnsahMohammad** [reply]
@choonho , finally, its done. 

![image](https://github.com/cloudforet-io/cloudforet/assets/85027826/b170a2bb-e726-4d52-bbfa-56fd8266fa31)

I think there's no need for the mentoring session then. Thanks a lot! 
Btw, there's no option on week 1 to upload now. Is it because the due date has passed?

**Reply by choonho** [reply]
Good Job. I will check your status to **completed** for task 1. Let's cancel the 1:1 meeting.

**Comment by SammyBloom** [comment]
Hi @choonho,
I have been unable to complete the Week 1 task. 
Here are some changes I made but I am still getting errors.
I atarted minikube with 3GB instead of 4GB.
![Screenshot 2023-12-11 185442](https://github.com/cloudforet-io/cloudforet/assets/32566612/473fa9aa-6119-434b-9a94-72749d6177a3)
![Screenshot 2023-12-11 185503](https://github.com/cloudforet-io/cloudforet/assets/32566612/bad061cc-2a03-41cd-bb2f-97d4f129d578)

**Reply by choonho** [reply]
@SammyBloom You did almost well. I think the problem is memory size. I also tested with 3GB. Installation failed lack of resources.

I will update your status as **completed**. Let's do Task 2.

**Reply by ATREAY** [reply]
Hey @choonho, I'm stuck on this step as well. If it's doable, could you mark my status as complete? I've been on this step for the past few weeks. Check the details [here](https://github.com/orgs/cloudforet-io/discussions/133#discussioncomment-7692232).

**Reply by SammyBloom** [reply]
Thanks so much @choonho
